### PR TITLE
WIP - GUI: Implement external cmdline

### PR DESCRIPTION
**Pull request by equalsraf**
Nothing to see here yet, just working out the api.

- [ ] I think this is api lvl 3, but I have not checked. the attachment code needs to be updated and we need to generate bindings.

ref https://github.com/equalsraf/neovim-qt/issues/347

### CODE REVIEW

**Comment by gagbo** [comment]
tl;dr : should we actually implement the cmdline as a sub-`Shell` widget ?

At first I thought that using a [QTextEdit](http://doc.qt.io/qt-5/qtextedit.html#details) widget would be the best solution for the issue, because QTextEdit can show/edit Rich Text. Rich text support is important to do something with the formatting information, contrary to [QLineEdit](http://doc.qt.io/qt-5/qlineedit.html#details).

Another advantage of QTextEdit over QLineEdit is the ability to show multiple lines/paragraphs. QLineEdit is not able to properly answer `cmdline_block_show` and `cmdline_block_append`

But then I thought that if we're editing text in this window we probably want a sub `Shell` there don't we ? And the `wildmenu` should then be implemented as a special instance of `PopupMenu` ?

**Comment by jgehrig** [comment]
Superseded by Pull #492 and Pull #586.

Marking as closed.
